### PR TITLE
Use ``python-typing-update`` on ``pylint/pyreverse`` directory

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Ref #4683.

@Pierre-Sassoulas @cdce8p Perhaps comment if you're going to review as not to duplicate work.

### CODE REVIEW

File: pylint/message/message.py [hunk 1]
```diff
@@ -41,33 +43,30 @@ def __new__(
...
 
     def __new__(
         cls,
         msg_id: str,
         symbol: str,
-        location: Union[
-            Tuple[str, str, str, str, int, int],
-            MessageLocationTuple,
-        ],
+        location: (tuple[str, str, str, str, int, int] | MessageLocationTuple),
```
Comment (Pierre-Sassoulas) [inline pylint/message/message.py:66]:
```suggestion
        location: MessageLocationTuple,
```
Could we do this ? It seems those should be the same thing.

Comment (DanielNoord) [reply pylint/message/message.py:66]: No, see the deprecation warning a couple of lines below. `location` needs to be initialised as `MessageLocationTuple` and not be a regular `tuple`.

File: pylint/message/message.py [hunk 2]
```diff
@@ -41,33 +43,30 @@ def __new__(
...
+    ) -> Message:
         ...
 
     @overload
     def __new__(
         cls,
         msg_id: str,
         symbol: str,
-        location: Tuple[str, str, str, str, int, int],
+        location: tuple[str, str, str, str, int, int],
```
Comment (Pierre-Sassoulas) [inline pylint/message/message.py:55]:
```suggestion
        location: MessageLocationTuple,
```
(🤞 ?)

**Review by Pierre-Sassoulas — APPROVED** [review]

File: pylint/message/message.py (hunk 1 above)
Comment (cdce8p) [inline pylint/message/message.py:66]: The brackets could have been removed here though.
```diff
-    location: (tuple[str, str, str, str, int, int] | MessageLocationTuple),
+    location: tuple[str, str, str, str, int, int] | MessageLocationTuple,
```

Comment (DanielNoord) [reply pylint/message/message.py:66]: Added in follow-up!

**Pull request merged**